llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -858,623 +858,433 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
 }
 
 const char *NVPTXTargetLowering::getTargetNodeName(unsigned Opcode) const {
+
+#define MAKE_CASE(V)                                                           \
```

Comment:
I wonder if we could do it the way WebAssembly does: https://github.com/llvm/llvm-project/blob/ab4a793e8bc78f50f9f104c9c732e2dd91bf70a2/llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp#L719

This approach is fine, but it's still too much of a boilerplate.

Reply:
Interesting. Something to consider for sure.